api/service/service.go

```diff
@@ -1,19 +1,68 @@
 package service
 
 import (
+	"context"
+	"time"
+
 	"github.com/gin-gonic/gin"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/types/known/emptypb"
 	"gorm.io/gorm"
+
+	"github.com/photon-storage/go-photon/sak/io/rpc"
+	pbc "github.com/photon-storage/photon-proto/consensus"
+	pbd "github.com/photon-storage/photon-proto/depot"
 )
 
 // Service defines an instance of service that handles third-party requests.
 type Service struct {
-	db *gorm.DB
+	ctx              context.Context
+	db               *gorm.DB
+	depotPk          []byte
+	depotDiscoveryID []byte
+	nodeCli          pbc.NodeClient
+	depotCli         pbd.DepotClient
 }
 
 // New creates a new service instance.
-func New(db *gorm.DB) *Service {
+func New(
+	ctx context.Context,
+	db *gorm.DB,
+	nodeEndpoint,
+	depotEndpoint string,
```

Comment:
Lets discover the depot through p2p network, instead of hard-coding its endpoint

https://github.com/photon-storage/go-photon/blob/main/cmd/localnet/object_commit.go#L81

To generalize even more, we should have a class to index depot and filter/select depot based on some criteria.

Reply:
solve this issue in next pr.